hexa/databases/schema.py

```diff
@@ -38,6 +40,35 @@ def resolve_database_table(workspace, info, **kwargs):
     return get_table_definition(workspace, kwargs.get("name"))
 
 
+@database_object.field("port")
+def resolve_database_port(_, info, **kwargs):
+    return get_db_server_credentials()["port"]
+
+
+@database_object.field("host")
+def resolve_database_host(_, info, **kwargs):
+    return get_db_server_credentials()["host"]
+
+
+@database_object.field("password")
+def resolve_database_password(workspace, info, **kwargs):
+    request: HttpRequest = info.context["request"]
+    return (
+        workspace.db_password
+        if request.user.has_perm("workspaces.manage_members", workspace)
```

Comment:
The permission is not the right one. You can just check if the user is an admin of the workspace